src/rpcserver.cpp

```diff
@@ -0,0 +1,205 @@
+                }
+                else
+                {
+                    Send ( pSocket, QJsonDocument ( CreateJsonRpcErrorReply ( -32600, "Invalid Request" ) ) );
+                }
+            }
+            else if ( data.isArray() )
+            {
+                QJsonArray output;
+                for ( auto item : data.array() )
+                {
+                    if ( item.isObject() )
+                    {
+                        auto        object = item.toObject();
+                        QJsonObject response;
+                        response["jsonrpc"] = QJsonValue ( "2.0" );
+                        response["id"]      = object["id"];
+                        ProcessMessage ( pSocket, object, response );
+                        output.append ( response );
+                    }
+                    else
+                    {
+                        output.append ( CreateJsonRpcErrorReply ( -32600, "Invalid Request" ) );
+                    }
+                }
+                if ( output.size() > 0 )
+                {
+                    Send ( pSocket, QJsonDocument ( output ) );
+                }
+                else
+                {
+                    Send ( pSocket, QJsonDocument ( CreateJsonRpcErrorReply ( -32600, "Invalid Request" ) ) );
+                }
+            }
+            else if ( data.isObject() )
+            {
+                auto        object = data.object();
+                QJsonObject response;
+                response["jsonrpc"] = QJsonValue ( "2.0" );
+                response["id"]      = object["id"];
+                ProcessMessage ( pSocket, object, response );
+                Send ( pSocket, QJsonDocument ( response ) );
+            }
+            else
+            {
+                Send ( pSocket, QJsonDocument ( CreateJsonRpcErrorReply ( -32600, "Invalid Request" ) ) );
+            }
+        }
+    } );
+}
+
+void CRpcServer::Send ( QTcpSocket* pSocket, const QJsonDocument& aMessage ) { pSocket->write ( aMessage.toJson ( QJsonDocument::Compact ) + "\n" ); }
+
+void CRpcServer::HandleMethod ( const QString& strMethod, CRpcHandler pHandler ) { mapMethodHandlers[strMethod] = pHandler; }
+
+void CRpcServer::ProcessMessage ( QTcpSocket* pSocket, QJsonObject message, QJsonObject& response )
+{
+    if ( !message["method"].isString() )
+    {
+        response["error"] = CreateJsonRpcError ( -32600, "Invalid Request" );
+        return;
+    }
+
+    // Obtain the method handler
+    auto method = message["method"].toString();
+    auto it     = mapMethodHandlers.find ( method );
+    if ( it == mapMethodHandlers.end() )
+    {
+        response["error"] = CreateJsonRpcError ( -32601, "Method not found" );
+        return;
+    }
+
+    // Obtain the params
+    auto jsonParams = message["params"];
+    if ( !jsonParams.isObject() )
+    {
+        response["error"] = CreateJsonRpcError ( -32602, "Invalid params: not an object" );
```

Comment:
This seems to be missing a `return`?

```suggestion
        response["error"] = CreateJsonRpcError ( -32602, "Invalid params: not an object" );
        return;
```